### PR TITLE
 [receiver/prometheus] Fix start timestamp when timestamp is present

### LINKED ISSUE #6405: [Prometheus Receiver] Incorrect Start_Timestamp for counter metric when explicit timestamp is present
***Describe the bug***

With `honor_timestamps` config set to true, the explicit timestamp read by the counter metric causes incorrect `start_timestamp` for the counter metric. 
In presence of explicit timestamp, counter metric `start_timestamp` is always set to the explicit timestamp provided to the first metric seen in the scrape. This `start_timestamp` also does not reset upon encountering counter value lower than previous scrape. 

***Steps to reproduce***

* Run OTEL Prometheus Receiver with `honor_timestamps` bool set to true in `ScrapeConfig` of Prometheus Receiver config. 
* Ensure that Prometheus Receiver scrapes the below testdata (provided with explicit timestamps). Provided explicit timestamps may need to be revised as per current time, as Prometheus scrape loop rejects data that is too old or too far
* Inspect the Prometheus Receiver output.

```
# HELP go_threads Number of OS threads created
# TYPE go_thread gauge
go_threads 19 1637604687818
# HELP http_requests_total The total number of HTTP requests.
# TYPE http_requests_total counter
http_requests_total{method="post",code="200"} 100 1637604697818
http_requests_total{method="post",code="400"} 5 1637604707818
# HELP http_request_duration_seconds A histogram of the request duration.
# TYPE http_request_duration_seconds histogram
http_request_duration_seconds_bucket{le="0.05"} 1000 1637604717818
http_request_duration_seconds_bucket{le="0.5"} 1500 1637604717818
http_request_duration_seconds_bucket{le="1"} 2000 1637604717818
http_request_duration_seconds_bucket{le="+Inf"} 2500 1637604717818
http_request_duration_seconds_sum 5000 1637604717818
http_request_duration_seconds_count 2500 1637604717818
# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
# TYPE rpc_duration_seconds summary
rpc_duration_seconds{quantile="0.01"} 1 1637604727818
rpc_duration_seconds{quantile="0.9"} 5 1637604727818
rpc_duration_seconds{quantile="0.99"} 8 1637604727818
rpc_duration_seconds_sum 5000 1637604727818
rpc_duration_seconds_count 1000 1637604727818
```

**What did you expect to see?**

Expected to see `start_timestamp` of counter metric equal to the `point_timestamp` and the explicit timestamp provided in the testdata, for the initial scrape. And for the consecutive scrapes, where the counter value is less than previous scrape, the counter `start_timestamp` should reset.

**Additional context**
Related to https://github.com/open-telemetry/wg-prometheus/issues/57
Linked Issue: #6085

cc @alolita @Aneurysm9

### PR DESCRIPTION
**Pull request by PaurushGarg**
**Description:** 

- Changing `mg.intervalStartTimeMs` to `mg.ts` for setting `start_timestamp` in `metricfamily` (to*TimeSeries), because we have `metric_adjuster` down the line in Prometheus Receiver that sets (and adjusts) the `start_timestamp`. 
- Fixing Issue 6405
- Removing the skipped logic of Issue 6085 (PR 6705).

**Link to tracking Issue:** 
Fixes #6405

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: receiver/prometheusreceiver/internal/otlp_metricfamily_test.go [hunk 1]
```diff
@@ -161,8 +161,8 @@ func TestMetricGroupData_toSummaryUnitTest(t *testing.T) {
 						{Name: "a", Value: "A"}, {Name: "quantile", Value: "0.0"}, {Name: "b", Value: "B"},
 					},
 					scrapes: []*scrape{
-						{at: 10, value: 10, metric: "histogram_count"},
-						{at: 10, value: 12, metric: "histogram_sum"},
+						{at: 10, value: 10, metric: "summary_count"},
```
Comment (codeboten) [inline receiver/prometheusreceiver/internal/otlp_metricfamily_test.go:164]: Why the metric name changes here? Was histogram incorrect?

File: receiver/prometheusreceiver/internal/otlp_metricfamily_test.go [hunk 2]
```diff
@@ -227,8 +227,8 @@ func TestMetricGroupData_toSummaryUnitTest(t *testing.T) {
 				qn99 := qtL.AppendEmpty()
 				qn99.SetQuantile(.99)
 				qn99.SetValue(82)
-				point.SetTimestamp(14 * 1e6) // the time in milliseconds -> nanoseconds.
-				point.SetStartTimestamp(10 * 1e5)
+				point.SetTimestamp(14 * 1e6)      // the time in milliseconds -> nanoseconds.
+				point.SetStartTimestamp(14 * 1e6) // the time in milliseconds -> nanoseconds
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/internal/otlp_metricfamily_test.go:231]: Use a constant for the 1e6 that better suggest the value.

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go [hunk 3]
```diff
@@ -176,7 +176,7 @@ func (mg *metricGroupPdata) toDistributionPoint(orderedLabelKeys []string, dest
 	// The timestamp MUST be in retrieved from milliseconds and converted to nanoseconds.
 	tsNanos := pdataTimestampFromMs(mg.ts)
 	if mg.family.isCumulativeTypePdata() {
-		point.SetStartTimestamp(pdataTimestampFromMs(mg.intervalStartTimeMs))
+		point.SetStartTimestamp(pdataTimestampFromMs(mg.ts)) // metrics_adjuster adjusts the startTimestamp to the initial scrape timestamp
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:179]: Is the intervalStartTimeMs still used? Can you cache the `pdataTimestampFromMs(mg.ts)`?

File: receiver/prometheusreceiver/internal/otlp_metricfamily_test.go (hunk 1 above)
Comment (PaurushGarg) [reply receiver/prometheusreceiver/internal/otlp_metricfamily_test.go:164]: @codeboten the method name is `TestMetricGroupData_toSummaryUnitTest(t *testing.T)` and it is testing `summary` metric, but the `metric` string was mentioned as `histogram`, so I corrected it to `summary`.

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go (hunk 3 above)
Comment (PaurushGarg) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go]: @bogdandrutu thanks. Updated. I have removed `intervalStartTimeMs` as it was no longer required. I have also used the cached `pdataTimestampFromMs(mg.ts)`

File: receiver/prometheusreceiver/internal/otlp_metricfamily_test.go (hunk 2 above)
Comment (PaurushGarg) [reply receiver/prometheusreceiver/internal/otlp_metricfamily_test.go]: @bogdandrutu thanks. I have updated.

**Comment by jpkrohling** [comment]
I'm waiting for @bogdandrutu, as he had comments about this PR.

**Review by codeboten — APPROVED** [review]

**Comment by alolita** [comment]
@bogdandrutu can you please merge this PR since it has two approvals from experts. Thx.

**Pull request merged**